Document advanced binding options

Srivaralakshmi opened the pull request:
- **Aligned team**: Dev Tools
- **OCP version for cherry-picking**: enterprise-`4.9`
- **JIRA issues**: [RHDEVDOCS-2824](https://issues.redhat.com/browse/RHDEVDOCS-2824), [RHDEVDOCS-3210](https://issues.redhat.com/browse/RHDEVDOCS-3210), [RHDEVDOCS-3212](https://issues.redhat.com/browse/RHDEVDOCS-3212), [RHDEVDOCS-3211](https://issues.redhat.com/browse/RHDEVDOCS-3211)
- **Preview pages**: [Binding workloads using Service Binding Operator](https://deploy-preview-39839--osdocs.netlify.app/openshift-enterprise/latest/applications/connecting_applications_to_services/binding-workloads-using-sbo.html)
- **SME Review**: Approved by @baijum, @pedjak
- **QE review**: Approved by @pmacik 
- **Peer-review**: Approved by @bburt-rh

Srivaralakshmi commented:
Hi @pedjak, @baijum, and @pmacik: PTAL at this PR and provide your feedback. Thanks!

bburt-rh commented on applications/connecting_applications_to_services/binding-workloads-using-sbo.adoc:
```diff
@@ -0,0 +1,47 @@
+
+toc::[]
+
+Application developers must bind a workload to one or more backing services using a binding secret generated for the purpose of storing information to be consumed by the workload.
```
```suggestion
Application developers must bind a workload to one or more backing services by using a binding secret. This secret is generated for the purpose of storing information to be consumed by the workload.
```
This sentence feels like there's a bit too much information in it. Consider creating two sentences for the content. Above is one suggestion for this edit.

bburt-rh commented on applications/connecting_applications_to_services/binding-workloads-using-sbo.adoc:
```diff
@@ -0,0 +1,47 @@
+
+Application developers must bind a workload to one or more backing services using a binding secret generated for the purpose of storing information to be consumed by the workload.
+
+As an example, consider that the service you want to connect to is already exposing the binding data. You also need a workload to be used along with the `ServiceBinding` custom resource (CR). Through this `ServiceBinding` CR the workload sends a binding request with the details of the services to bind with.
```
```suggestion
As an example, consider that the service you want to connect to is already exposing the binding data. In this case, you would also need a workload to be used along with the `ServiceBinding` custom resource (CR). Bu using this `ServiceBinding` CR, the workload sends a binding request with the details of the services to bind with.
```
- Use the conditional tense here for a hypothetical example.
- A couple of other small suggestions

bburt-rh commented on applications/connecting_applications_to_services/binding-workloads-using-sbo.adoc:
```diff
@@ -0,0 +1,47 @@
+      resource: deployments
+----
+<1> Specifies a list of service resources.
+<2> The sample application that points to a Deployment or any other similar resource with an embedded PodSpec.
```
```suggestion
<2> The sample application that points to a Deployment or any other similar resource with an embedded PodSpec resource.
```

Srivaralakshmi replied:
It is just the embedded Podspec. I have removed the resource as that was a typo..

bburt-rh commented on applications/connecting_applications_to_services/binding-workloads-using-sbo.adoc:
```diff
@@ -0,0 +1,47 @@
+----
+<1> Specifies a list of service resources.
+<2> The sample application that points to a Deployment or any other similar resource with an embedded PodSpec.
+resource.
```
```suggestion
```

Srivaralakshmi replied:
removed this.

bburt-rh commented on modules/sbo-advanced-binding-options.adoc:
```diff
@@ -0,0 +1,113 @@
+[id="sbo-advanced-binding-options_{context}"]
+= Advanced binding options
+
+This section describes the advanced binding options, which are available only for the `binding.operators.coreos.com` API group.
```
```suggestion
Advanced binding options are available only for the `binding.operators.coreos.com` API group.
```
I'm not sure you really need the "this section describes" phrasing, which doesn't really add any meaningful content. Your call on whether you think it's necessary.

Srivaralakshmi commented on modules/sbo-advanced-binding-options.adoc:
```diff
@@ -0,0 +1,113 @@
+This section describes the advanced binding options, which are available only for the `binding.operators.coreos.com` API group.
+
+=== Changing the binding names before projecting into workload
+You can specify the rules to change the binding names in the .spec.namingStrategy attribute of the `ServiceBinding` custom resource (CR). For example, consider a Spring PetClinic sample application, which connects to the PostgreSQL database and the host and port fields of the database are exposed by the the PostgreSQL database service to use for binding. The sample application can access this bin ...
```
I have addressed some suggestions and have revised the content. I have ignored some suggestions to retain the technical accuracy.

Srivaralakshmi commented:
@bburt-rh Thanks for the suggestions. Very helpful! I have addressed most of them. PTAL and provide feedback.

bburt-rh commented on modules/sbo-advanced-binding-options.adoc:
```diff
@@ -0,0 +1,113 @@
+=== Composing custom binding data
+As an application developer, you can compose binding data under the following circumstances:
+
+* The backing service does not expose binding data
```
```suggestion
* The backing service does not expose binding data.
```

bburt-rh commented on modules/sbo-advanced-binding-options.adoc:
```diff
@@ -0,0 +1,113 @@
+* The backing service does not expose binding data
+* The values exposed are not available in the required format as expected by the workload.
+
+For example, consider a case where the backing service CR exposes the host, port, and database user as binding data, but the workload requires to consume this binding data as a connection string.
```
```suggestion
For example, consider a case where the backing service CR exposes the host, port, and database user as binding data, but the workload requires that the binding data be consumed as a connection string.
```

bburt-rh commented on modules/sbo-naming-strategies.adoc:
```diff
@@ -0,0 +1,51 @@
+[id="sbo-naming-strategies_{context}"]
+= Naming strategies
+
+This section describes the naming strategies, which are available only for the `binding.operators.coreos.com` API group.
```
```suggestion
Naming strategies are available only for the `binding.operators.coreos.com` API group.
```
Remove `This section describes`. It doesn't really add anything meaningful, and anyway the title of the section implies what the content will be.

bburt-rh commented on modules/sbo-naming-strategies.adoc:
```diff
@@ -0,0 +1,51 @@
+
+This section describes the naming strategies, which are available only for the `binding.operators.coreos.com` API group.
+
+Naming strategies use Go templates to help you define custom binding names through the service binding request. Naming strategies are applicable for all attributes including the mappings in the `ServiceBinding` custom resource (CR).
```
```suggestion
Naming strategies use Go templates to help you define custom binding names through the service binding request. Naming strategies apply for all attributes including the mappings in the `ServiceBinding` custom resource (CR).
```
Edit for concision

bburt-rh commented on modules/sbo-naming-strategies.adoc:
```diff
@@ -0,0 +1,51 @@
+While using the naming strategies, depending on the expectations or requirements of your workload, you can use the following predefined post-processing functions in any combination to convert the character strings:
+
+* `upper`: Converts the character strings into capital or uppercase letters
+* `lower`: Converts the character strings into lowercase letters
```
```suggestion
* `lower`: Converts the character strings into lowercase letters.
```
Either use periods or no periods consistently in the bulleted list

bburt-rh commented on modules/sbo-naming-strategies.adoc:
```diff
@@ -0,0 +1,51 @@
+.Predefined post-processing functions
+While using the naming strategies, depending on the expectations or requirements of your workload, you can use the following predefined post-processing functions in any combination to convert the character strings:
+
+* `upper`: Converts the character strings into capital or uppercase letters
```
```suggestion
* `upper`: Converts the character strings into capital or uppercase letters.
```

bburt-rh commented on modules/sbo-naming-strategies.adoc:
```diff
@@ -0,0 +1,51 @@
+
+[NOTE]
+====
+* When the binding names are projected as files, by default the predefined `none` naming strategy is applied and there are no changes in the binding names.
```
```suggestion
* When the binding names are projected as files, by default the predefined `none` naming strategy is applied, and there are no changes in the binding names.
```
Need comma before a coordinating conjunction.

pmacik commented:
Doesn't a `workload` actually mean an `application`?

pmacik commented on applications/connecting_applications_to_services/binding-workloads-using-sbo.adoc:
```diff
@@ -0,0 +1,46 @@
+[id="binding-workloads-using-sbo"]
+= Binding workloads using Service Binding Operator
```
Doesn't a `workload` actually mean an `application`?

pmacik replied:
There seems to be a confusion in terms here

Srivaralakshmi replied:
@pmacik Yes, but when we are describing in general it was decided in our team SBO call meeting that we will use the term workload. And while explaining specific examples and uses cases we will use the term application.

baijum reviewed (approved):
LGTM

pmacik commented:
LGTM

bburt-rh commented on modules/sbo-advanced-binding-options.adoc:
```diff
@@ -0,0 +1,113 @@
+...
+----
+
+Consider that `namingStrategy` is not defined and the binding names are projected as environment variables. Then the `host: hippo-pgbouncer` value in the backing service and the projected environment variable appears as shown in the following example:
```
```suggestion
Consider that `namingStrategy` is not defined and the binding names are projected as environment variables. Then, the `host: hippo-pgbouncer` value in the backing service and the projected environment variable would appear as shown in the following example:
```
Use conditional for hypthetical example

bburt-rh commented on modules/sbo-advanced-binding-options.adoc:
```diff
@@ -0,0 +1,113 @@
+
+Following are the descriptions of the expressions defined in the `POSTGRESQL_{{ .service.kind | upper }}_{{ .name | upper }}_ENV` naming strategy:
+
+* `.name`: Refers to the binding name exposed by the backing service. In the case of previous example, the binding names are `HOST` and `PORT`.
```
```suggestion
* `.name`: Refers to the binding name exposed by the backing service. In the case of the previous example, the binding names are `HOST` and `PORT`.
```

bburt-rh commented on modules/sbo-binding-workloads-that-are-not-compliant-with-PodSpec.adoc:
```diff
@@ -0,0 +1,151 @@
+[id="sbo-binding-workloads-that-are-not-compliant-with-PodSpec_{context}"]
+= Binding secondary workloads that are not compliant with PodSpec
+
+A typical scenario in service binding involves backing service, workload (Deployment), and {servicebinding-title}. Consider a scenario that involves a secondary workload, which can also be an application Operator, not compliant with PodSpec and is between the primary workload (Deployment) and {servicebinding-title}.
```
```suggestion
A typical scenario in service binding involves configuring the backing service, the workload (Deployment), and {servicebinding-title}. Consider a scenario that involves a secondary workload (which can also be an application Operator) that is not compliant with PodSpec and is between the primary workload (Deployment) and {servicebinding-title}.
```
This is a little bit confusing to me as it's currently worded. I've made a few suggestions, but feel free to modify as needed if I've changed the meaning too much,

Srivaralakshmi replied:
Good catch! Thanks a lot for the suggestions.

bburt-rh commented on modules/sbo-binding-workloads-that-are-not-compliant-with-PodSpec.adoc:
```diff
@@ -0,0 +1,151 @@
+
+A typical scenario in service binding involves backing service, workload (Deployment), and {servicebinding-title}. Consider a scenario that involves a secondary workload, which can also be an application Operator, not compliant with PodSpec and is between the primary workload (Deployment) and {servicebinding-title}.
+
+For such secondary workload resources, the location of the container path is arbitrary. For service binding, if the secondary workload in a CR is not compliant with the PodSpec, you must specify the location of the container path. This helps in projecting the binding data into the container path specified in the secondary workload of the `ServiceBinding` CR, for example, when you do not want the  ...
```
```suggestion
For such secondary workload resources, the location of the container path is arbitrary. For service binding, if the secondary workload in a CR is not compliant with the PodSpec, you must specify the location of the container path. Doing so projects the binding data into the container path specified in the secondary workload of the `ServiceBinding` CR, which you might want to do if you do not want the binding data inside a pod.
```
- `This helps` - What does `This` refer to in this sentence? There's no direct antecedent, so I'm not sure what the sentence is referring to. See the "Grammar" section of the ISG for more information: e.g. `Make the subject of a sentence clear.`
- `helps` doesn't seem like the right word to use if the specifying the location of the container path is required.
- My edit shows one possible way to clarify the relationship between the different pieces of information, though maybe it alters the meaning too much.

Srivaralakshmi replied:
Yes, agree, thanks for pointing this out. I have implemented "Doing so projects". Retaining the last sentence as is as changing this to  "Which you can do if" alters the meaning.

bburt-rh commented on modules/sbo-binding-workloads-that-are-not-compliant-with-PodSpec.adoc:
```diff
@@ -0,0 +1,151 @@
+{servicebinding-title} provides an option to configure the value of where the container path or secret path is and bind these paths at a custom location. This option is available only for the `binding.operators.coreos.com` API group when the binding data is projected as environment variables.
+
+=== Configuring the custom location of the container path
+Consider a secondary workload CR, which is not compliant with the PodSpec, with containers at the `spec.containers` path:
```
```suggestion
Consider a secondary workload CR that is not compliant with the PodSpec and has containers located at the `spec.containers` path:
```

Srivaralakshmi replied:
Yes, you are correct. 100%

This section is specifically for the secondary workloads that are not compliant with PodSpec. Hence I am retaining ", which" everywhere and ignoring your suggestions.

bburt-rh commented on modules/sbo-binding-workloads-that-are-not-compliant-with-PodSpec.adoc:
```diff
@@ -0,0 +1,151 @@
+      - containerPort: 9966
+----
+
+The following example shows how to configure the value of where the `spec.containers` path is by specifying it in the `ServiceBinding` CR and bind this path at a `spec.application.bindingPath.containersPath` custom location:
```
```suggestion
The following example shows how to configure the `spec.containers` path by specifying a value in the `ServiceBinding` CR and to bind this path to a `spec.application.bindingPath.containersPath` custom location:
```
The syntax here is a bit wordy and confusing. The suggested edit (I hope) clarifies the meaning, though I hope it doesn't alter that meaning.

Srivaralakshmi replied:
Absolutely, This looks so much better!! Thanks, @bburt-rh :)

bburt-rh commented on modules/sbo-binding-workloads-that-are-not-compliant-with-PodSpec.adoc:
```diff
@@ -0,0 +1,151 @@
+        containersPath: spec.containers <3>
+----
+<1> The sample application that points to a Deployment or any other similar resource with an embedded PodSpec.
+<2> Secondary workload, which is not compliant with the PodSpec.
```
```suggestion
<2> The secondary workload, which is not compliant with the PodSpec.
```

bburt-rh commented on modules/sbo-binding-workloads-that-are-not-compliant-with-PodSpec.adoc:
```diff
@@ -0,0 +1,151 @@
+----
+<1> The sample application that points to a Deployment or any other similar resource with an embedded PodSpec.
+<2> Secondary workload, which is not compliant with the PodSpec.
+<3> Custom location of the container path.
```
```suggestion
<3> The custom location of the container path.
```

bburt-rh commented on modules/sbo-binding-workloads-that-are-not-compliant-with-PodSpec.adoc:
```diff
@@ -0,0 +1,151 @@
+
+After you specify the location of the container path, {servicebinding-title} generates the binding data, which becomes available in the container path specified in the secondary workload of the `ServiceBinding` CR.
+
+The following example shows the `spec.containers` path, which is updated  with the `envFrom` and `secretRef` fields:
```
```suggestion
The following example shows the `spec.containers` path with the `envFrom` and `secretRef` fields:
```
The passive voice in the relative clause at the end makes this a bit confusing. What or who is actually updating these fields? It's not clear to me. My edit might help, but maybe it changes the meaning too much

Srivaralakshmi replied:
The fields are updated by the Service Binding Operator. But I like the revision so going with your suggestion.

bburt-rh commented on modules/sbo-binding-workloads-that-are-not-compliant-with-PodSpec.adoc:
```diff
@@ -0,0 +1,151 @@
+<2> Name of the `Secret` resource generated by the {servicebinding-title}.
+
+=== Configuring the custom location of the secret path
+Consider a secondary workload CR, which is not compliant with the PodSpec, with only the secret at the `spec.secret` path:
```
```suggestion
Consider a secondary workload CR that is not compliant with the PodSpec with only the secret at the `spec.secret` path:
```
Again, that relative clause confuses me. It makes it sound like all secondary workload CRs are not compliant.

Srivaralakshmi replied:
Yes, it is correct. so ignoring your revision here.

bburt-rh commented on modules/sbo-binding-workloads-that-are-not-compliant-with-PodSpec.adoc:
```diff
@@ -0,0 +1,151 @@
+    secret: ""
+----
+
+The following example shows how to configure the value of where the `spec.secret` path is by specifying it in the `ServiceBinding` CR and bind this path at a `spec.application.bindingPath.secretPath` custom location:
```
```suggestion
The following example shows how to configure `spec.secret` path by specifying a value in the `ServiceBinding` CR and to bind this path at a `spec.application.bindingPath.secretPath` custom location:
```

bburt-rh commented on modules/sbo-binding-workloads-that-are-not-compliant-with-PodSpec.adoc:
```diff
@@ -0,0 +1,151 @@
+        secretPath: spec.secret <2>
+...
+----
+<1> Secondary workload, which is not compliant with the PodSpec.
```
```suggestion
<1> Secondary workload that is not compliant with the PodSpec.
```

Srivaralakshmi replied:
Taking the 1st part, but ignoring the latter part :)

bburt-rh commented on modules/sbo-binding-workloads-that-are-not-compliant-with-PodSpec.adoc:
```diff
@@ -0,0 +1,151 @@
+...
+----
+<1> Secondary workload, which is not compliant with the PodSpec.
+<2> Custom location of the secret path that contains the name of the `Secret` resource.
```
```suggestion
<2> The custom location of the secret path that contains the name of the `Secret` resource.
```

bburt-rh commented on modules/sbo-binding-workloads-that-are-not-compliant-with-PodSpec.adoc:
```diff
@@ -0,0 +1,151 @@
+
+After you specify the location of the secret path, {servicebinding-title} generates the binding data, which becomes available in the secret path specified in the secondary workload of the `ServiceBinding` CR.
+
+The following example shows the `spec.secret` path, which is updated  with the `binding-request` value:
```
```suggestion
The following example shows the `spec.secret` path with the `binding-request` value:
```

bburt-rh commented on modules/sbo-naming-strategies.adoc:
```diff
@@ -0,0 +1,51 @@
+
+Naming strategies use Go templates to help you define custom binding names through the service binding request. Naming strategies apply for all attributes including the mappings in the `ServiceBinding` custom resource (CR).
+
+A backing service projects the binding names as files or environment variables into the workload. If a workload expects the projected binding names in a particular format, but the binding names to be projected from the backing service are not available in that format, then you can change the binding names using the naming strategies.
```
```suggestion
A backing service projects the binding names as files or environment variables into the workload. If a workload expects the projected binding names in a particular format, but the binding names to be projected from the backing service are not available in that format, then you can change the binding names using naming strategies.
```

bburt-rh commented on modules/sbo-naming-strategies.adoc:
```diff
@@ -0,0 +1,51 @@
+
+[NOTE]
+====
+* When the binding names are projected as files, by default the predefined `none` naming strategy is applied, and there are no changes in the binding names.
```
```suggestion
* When the binding names are projected as files, by default the predefined `none` naming strategy is applied, and the binding names do not change.
```
Per ISG: avoide  "there is" and "there are" constructions. See the "Grammar" section for more info,

Srivaralakshmi replied:
Perfect! Thanks for the reference. Looks good now :)

bburt-rh commented on modules/sbo-unbinding-workloads-from-a-backing-service.adoc:
```diff
@@ -0,0 +1,24 @@
+
+You can unbind a workload from a backing service by using the `oc` tool.
+
+. To unbind a workload from a backing service, delete the `ServiceBinding` custom resource (CR) linked to it :
```
```suggestion
. To unbind a workload from a backing service, delete the `ServiceBinding` custom resource (CR) linked to it:
```

Srivaralakshmi replied:
Good catch! Thanks a lot @bburt-rh  :)

bburt-rh commented on modules/sbo-unbinding-workloads-from-a-backing-service.adoc:
```diff
@@ -0,0 +1,24 @@
+You can unbind a workload from a backing service by using the `oc` tool.
+
+. To unbind a workload from a backing service, delete the `ServiceBinding` custom resource (CR) linked to it :
+
```
+

Srivaralakshmi replied:
Again, good catch! Thanks my savior :)

bburt-rh commented on modules/sbo-unbinding-workloads-from-a-backing-service.adoc:
```diff
@@ -0,0 +1,24 @@
+----
+$ oc delete ServiceBinding <.metadata.name>
+----
+
```
```suggestion
+
```

bburt-rh commented on modules/sbo-unbinding-workloads-from-a-backing-service.adoc:
```diff
@@ -0,0 +1,24 @@
+----
+
+.Example
+[source,yaml]
```
```suggestion
[source,terminal]
```

Srivaralakshmi replied:
Phew! I guess I was too exhausted by the end of the day. How can I miss this?

bburt-rh commented on modules/sbo-unbinding-workloads-from-a-backing-service.adoc:
```diff
@@ -0,0 +1,24 @@
+----
+$ oc delete ServiceBinding spring-petclinic-rest
+----
+where:
```
```suggestion
+
where:
```

bburt-rh commented on modules/sbo-unbinding-workloads-from-a-backing-service.adoc:
```diff
@@ -0,0 +1,24 @@
+
+[source,terminal]
+----
+$ oc delete ServiceBinding <.metadata.name>
```
```suggestion
$ oc delete ServiceBinding <name_of_ServiceBinding_CR>
```
See the SSG section on "User-replaced values" for guidance on formatting. Also, if you change `.metadata.name` to `name_of_ServiceBinding_CR` then you can delete the whole `where` section below (as I have done)

Srivaralakshmi replied:
Nope, I would like to retain this because ".meatadata.name" is the attribute in the CR of the service binding resource.

Our OCP guideline has the provision for such rare usage needs. Refer [here](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output): "If you must provide additional information on what a line of a code block represents and the use of callouts is impractical, you can use a description list to provide information about the variables in the code block. Using callouts might be impractical if a code block contains too many conditional statements to easily use numbered callouts or if the same note applies to multiple lines of the codeblock."

I am retaining the content as is. Apologies for not being able to incorporate your suggestions. Thanks for your understanding and cooperation :)

bburt-rh replied:
@Srivaralakshmi Thanks for the clarification. I learned something new!

bburt-rh commented on modules/sbo-unbinding-workloads-from-a-backing-service.adoc:
```diff
@@ -0,0 +1,24 @@
+----
+$ oc delete ServiceBinding spring-petclinic-rest
+----
+where:
```
```suggestion

```

Srivaralakshmi replied:
Please see the reference I have given above to the OCP guidelines.

bburt-rh commented on modules/sbo-unbinding-workloads-from-a-backing-service.adoc:
```diff
@@ -0,0 +1,24 @@
+$ oc delete ServiceBinding spring-petclinic-rest
+----
+where:
+[horizontal]
```
```suggestion

```

bburt-rh commented on modules/sbo-unbinding-workloads-from-a-backing-service.adoc:
```diff
@@ -0,0 +1,24 @@
+----
+where:
+[horizontal]
+`spring-petclinic-rest`:: Specifies the name of the `ServiceBinding` CR.
```
```suggestion
```

bburt-rh replied:
```suggestion
where `spring-petclinic-rest` specifies the name of the `ServiceBinding` CR.
```
@Srivaralakshmi The formatting of this bit in the preview doesn't look right with the capital letter `Specifies`. Is there a reason why you can't just put the "where" sentence all in one line as I suggest here?

Srivaralakshmi commented:
Hi, @bburt-rh Thank you so very much for the review and useful comments and suggestions! I really appreciate the value add and positive difference you bring to the content with your quality reviews. Thanks again! 

I have addressed most of your comments. PTAL at the revised PR and provide your LGTM.

pedjak reviewed (approved):
LGTM

bburt-rh commented on modules/sbo-advanced-binding-options.adoc:
```diff
@@ -0,0 +1,113 @@
+
+Advanced binding options are available only for the `binding.operators.coreos.com` API group.
+
+=== Changing the binding names before projecting into workload
```
```suggestion
=== Changing the binding names before projecting into the workload
```
Does `workload` need a definite article here? I'm not sure how it's used in this product. You decide.

Srivaralakshmi replied:
Yes, retaining as is.

bburt-rh commented on modules/sbo-advanced-binding-options.adoc:
```diff
@@ -0,0 +1,113 @@
+...
+----
+
+Consider that `namingStrategy` is not defined and the binding names are projected as environment variables. Then, the `host: hippo-pgbouncer` value in the backing service and the projected environment variable would appear as shown in the following example:
```
```suggestion
If `namingStrategy` is not defined and the binding names are projected as environment variables, then the `host: hippo-pgbouncer` value in the backing service and the projected environment variable would appear as shown in the following example:
```
Very minor suggestion to make the "if, then" relationship more apparent.

bburt-rh commented on modules/sbo-advanced-binding-options.adoc:
```diff
@@ -0,0 +1,113 @@
+POSTGRESQL_DATABASE_PORT_ENV: 5432
+----
+
+Following are the descriptions of the expressions defined in the `POSTGRESQL_{{ .service.kind | upper }}_{{ .name | upper }}_ENV` naming strategy:
```
```suggestion
The following items describe the expressions defined in the `POSTGRESQL_{{ .service.kind | upper }}_{{ .name | upper }}_ENV` naming strategy:
```

bburt-rh commented on modules/sbo-binding-workloads-that-are-not-compliant-with-PodSpec.adoc:
```diff
@@ -0,0 +1,151 @@
+[id="sbo-binding-workloads-that-are-not-compliant-with-PodSpec_{context}"]
+= Binding secondary workloads that are not compliant with PodSpec
+
+A typical scenario in service binding involves configuring the backing service, the workload (Deployment), and {servicebinding-title}. Consider a scenario that involves a secondary workload (which can also be an application Operator) that is  not compliant with PodSpec and is between the primary workload (Deployment) and {servicebinding-title}.
```
```suggestion
A typical scenario in service binding involves configuring the backing service, the workload (Deployment), and {servicebinding-title}. Consider a scenario that involves a secondary workload (which can also be an application Operator) that is not compliant with PodSpec and is between the primary workload (Deployment) and {servicebinding-title}.
```

Srivaralakshmi commented:
@bburt-rh I have addressed all your comments. Thanks for the review and suggestions. Please take a final look.

bburt-rh commented:
@Srivaralakshmi  LGTM

Preeticp commented:
/cherrypick enterprise-4.9

Preeticp commented:
/cherrypick enterprise-4.10
